malaria-ios/Managers/NotificationManager.swift

```diff
@@ -3,52 +3,76 @@ import UIKit
-        super.init(context: context)
+    guard let data = localNotificationArrayData else {
+      return
     }
     
-    /// Schedule a notification at the fireTime given by argument
-    ///
-    /// All previous notifications will be unsheduled
-    ///
-    /// - parameter `NSDate`:: fireTime
-    public func scheduleNotification(fireTime: NSDate){
-        Logger.Info("Sheduling \(category) to " + fireTime.formatWith("dd-MMMM-yyyy hh:mm"))
-        let notification: UILocalNotification = createNotification(fireTime)
-        UIApplication.sharedApplication().scheduleLocalNotification(notification)
-    }
+    let scheduledNotifications = NSKeyedUnarchiver.unarchiveObjectWithData(data) as? [UILocalNotification]
     
-    /// Unschedule all notifications with the category
-    public func unsheduleNotification(){
-        for event in UIApplication.sharedApplication().scheduledLocalNotifications! {
-            if event.category == category {
-                UIApplication.sharedApplication().cancelLocalNotification(event)
-            }
-        }
+    guard let notifications = scheduledNotifications else {
+      return
     }
     
-    /// Creates a notification
-    ///
-    /// - parameter `NSDate`:: fireTime
-    /// - returns: `UILocalNotification`: local notification
-    private func createNotification(fireDate: NSDate) -> UILocalNotification{
-        let localNotification = UILocalNotification()
-        localNotification.fireDate = fireDate
-        localNotification.soundName = UILocalNotificationDefaultSoundName;
-        localNotification.alertAction = alertAction
-        localNotification.alertBody = alertBody
-        localNotification.soundName = UILocalNotificationDefaultSoundName
-        localNotification.category = category
-        
-        return localNotification
+    for notification in notifications
+      where notification.category == category {
+        UIApplication.sharedApplication().cancelLocalNotification(notification)
     }
+  }
+  
+  /// Creates a notification
+  ///
+  /// - parameter `NSDate`:: fireTime
+  /// - returns: `UILocalNotification`: local notification
+  private func createNotification(fireDate: NSDate) -> UILocalNotification{
+    let localNotification = UILocalNotification()
+    localNotification.fireDate = fireDate
+    localNotification.soundName = UILocalNotificationDefaultSoundName;
+    localNotification.alertAction = alertAction
+    localNotification.alertBody = alertBody
+    localNotification.soundName = UILocalNotificationDefaultSoundName
+    localNotification.category = category
+    
+    scheduledNotifications.append(localNotification)
+    
+    // Save the notifications persistently
+    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
+    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
```

Comment:
The UserSettingsManager only adds Strings and Bools. Should I add a method to add objects? (the local notification array)